docs/components/input-file/use-cases.md

```diff
@@ -0,0 +1,424 @@
+      .accept="${'.jpg,.svg,.xml,image/svg+xml'}"
+      @file-list-changed="${ev => {
+        console.log('fileList', ev.detail.newFiles);
+      }}"
+    >
+    </lion-input-file>
+  `;
+};
+```
+
+### Validation
+
+#### Accept
+
+The `accept` attribute value is a string that defines the file types the file input should accept. This string is a comma-separated list of unique file type specifiers.
+
+For more info please checkout: <https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#accept>.
+
+```js preview-story
+export const acceptValidator = () => {
+  return html`
+    <lion-input-file
+      accept=".jpg,.svg,.xml,image/svg+xml"
+      label="Upload"
+      enable-drop-zone
+      @file-list-changed="${ev => {
+        console.log(ev.detail.newFiles);
+      }}"
+    >
+    </lion-input-file>
+  `;
+};
+```
+
+#### Maximum File Size
+
+The `max-file-size` attribute sets the maximum file size in bytes.
+
+```js preview-story
+export const sizeValidator = () => {
+  return html`
+    <lion-input-file
+      max-file-size="2048"
+      label="Upload"
+      @file-list-changed="${ev => {
+        console.log(ev.detail.newFiles);
+      }}"
+    >
+    </lion-input-file>
+  `;
+};
+```
+
+### Multiple file upload
+
+When file has to be uploaded as soon as it is selected by the user. Use `file-list-changed` event to get the newly added files, upload it to your server and set the response back to component via `uploadResponse` property for each file.
+
+```js preview-story
+export const multipleFileUpload = () => {
+  return html`
+    <lion-input-file
+      label="Upload"
+      name="upload"
+      .multiple="${true}"
+      max-file-size="1024000"
+      @file-removed="${ev => {
+        console.log('removed file details', ev.detail);
+      }}"
+      @file-list-changed="${ev => {
+        console.log('file-list-changed', ev.detail.newFiles);
+      }}"
+      @model-value-changed="${ev => {
+        console.log('model-value-changed', ev);
+      }}"
+    >
+    </lion-input-file>
+  `;
+};
+```
+
+### Pre-filled state
+
+The pre-filled state of the file upload component shows the files that were being assigned by the user initially.
+
+On every reload of page, the file upload component reverts to the initial state and using the `uploadResponse` property, shows the names(not links) of previously assigned files.
+
+**Note:**
+
+- the list of prefilled files will not be available in `modelValue` or `serializedValue`. This is because value cannot be assigned to native `input` with type `file`
+- Only `Required` validation (if set on `lion-file-input`) is triggered for files which are prefilled.
+
+```js preview-story
+export const prefilledState = () => {
+  return html`
+    <lion-input-file
+      label="Upload"
+      name="myFiles"
+      .multiple="${true}"
+      .validators="${[new Required()]}"
+      .uploadResponse="${[
```

Comment:
Can we not just let the consumer only specify the filenames and we fill in the rest to make it easier for consumers?
What exactly is the purpose of prefilling? Are the specified files also uploaded somehow?